security/fma-kona-asterisc.md

```diff
@@ -0,0 +1,213 @@
+
+- **Mitigations:**
+  State testing against op-revm.
+  Action tests covering the L2 block executor are run on every pull request in the [kona] repository.
+  `op-reth` which uses the same op-execution environment as `kona`.
+
+- **Detection:**
+  Running `op-reth` so it's synced and following tip would provide a way to watch for chain splits.
+  A chain split occuring means `kona`'s execution (the same as `op-reth`) diverges from the `op-program`.
+
+- **Recovery Path(s):**
+  If `kona` + `asterisc` is in the coldpath, it would need to be fixed and re-deployed.
+  If `op-program` + `cannon` cannot be placed in the hotpath, and `kona` + `asterisc` fails, the permissioned fallback must be engaged.
+  [The "Kona and Asterisc Hot Path Failure"](#-Kona-and-Asterisc-Hot-Path-Failure) above covers the failure mode for `kona` + `asterisc` when placed in the hot path.
+
+
+### Breaking Backwards Compatibility for the op-challenger
+
+- **Description:**
+  The `op-challenger` has been updated to support the Kona + Asterisc Fault Dispute Game type ([Game Type 3]).
+
+- **Risk Assessment:**
+  Low likelihood as the action tests that are run in CI very frequently run the `op-challenger` with the `kona` + `asterisc` fault dispute game.
+  Medium impact since the off-chain `op-challenger` service can be updated with enough buffer time due to the large game time and nature of the chess clock.
+
+- **Mitigations:**
+  The `op-challenger` is architected in such a way as to isolate game types and more so the individual game players.
+  To allow the challenger to scale and play many different game types at once, a list of game types to play is passed into the `op-challenger` as a CLI flag.
+  When games are detected on-chain, the `op-challenger` will create a "game player" for this game type.
+  Players are run in individual threads so a breaking `kona` + `asterisc` game player in the challenger will not cause a liveness issue for the `op-challenger` to play other game types.
+  Another key mitigation is the VM runner. This is an offline simulation that runs asterisc + kona with the op-challenger. It's been running since around August 2024.
+
+- **Detection:**
+  The Dispute Monitor ([op-dispute-mon] service provides an isolated monitoring service that will detect if a game is not being played by the honest `op-challenger`.
+  It will alert in this case.
+
+- **Recovery Path(s):**
+  If the `op-challenger` fails to play `kona` + `asterisc` fault dispute games, a fix will need to be made to the `op-challenger`.
+  It does not require governance, but would need a new release of the `op-challenger` to be published and rolled out to infrastructure.
+
+[op-dispute-mon]: https://github.com/ethereum-optimism/optimism/tree/develop/op-dispute-mon
+[Game Type 3]: https://github.com/ethereum-optimism/optimism/blob/develop/op-challenger/game/fault/types/types.go#L30
+
+
+### Divergence between RISCV.sol and Asterisc
```

Comment:
NIT: It would be nice to add the link the current repository, that are doing the differential fuzzing and testing to ensure there is no difference between the computation.